pep-0432.txt

```diff
@@ -57,12 +60,37 @@ explored by way of private updates to the initialisation API used by the main
 CPython CLI application.
 
 As long as that is the case, the Discussions-To header in the PEP will remain
-set to ``capi-sig@python.org``. Once a coherent and feasible proposal for a new
-public API has been developed, then that PEP header will be removed, and the
-PEP will be submitted back to ``python-dev`` for further review and discussion.
-(Python 3.9 currently seems like a more plausible time frame for that than
-Python 3.8 - if that changes, then the target Python version in the PEP will be
-adjusted accordingly)
+set to ``capi-sig@python.org``.
+
+However, as coherent and feasible proposals to address subsets of the problem
+are identified, they may be spun out to as separate PEPs, which will then be
+added to this PEP as dependencies.
+
+So far, the following subproposals have been extracted:
+
+* PEP 587 (Python Initialization Configuration): this splits out the
+  preinitialization stage, but otherwise keeps the combined ``Py_Initialize*``
+  model that configureds a Python interpreter with a ready-to-run ``__main__``
+  module in a single step. While this is still a nice improvement over the
+  pre-PEP status quo and gets embedding applications back on an equal footing
+  with the private APIs exposed to the default CPython CLI in the Python 3.7
+  release, it doesn't yet allow for the concept of a "sans IO" CPython runtime
+  that relies entirely on the embedding application for access to operating
+  system interfaces, nor does it allow for more of the configuration code to
+  be migrated out of C and into frozen Python modules.
```

Comment:
I'm not sure sure of what does "sans IO" mean. PyConfig allows you to populate the "path config output" and so Python doesn't access the filesystem to compute these fields. If you set module_search_paths to an empty list (and use_module_search_paths must be set to 1 in that case), Python will not access the filesystem to import a module. importlib and its "path configuration" (sys.path) is the largest consumer of filesystem access during configuration.

But I would prefer to discuss "sans IO" on python-dev rather than here ;-)

Reply:
See my email there: https://mail.python.org/pipermail/python-dev/2019-April/156991.html

Reply:
I have a vague notion of an even more restricted variant of the runtime, where even the `os` module has been stripped of it's ability to do much of anything before `__main__` runs.

However, it's a distraction here, so I've changed that sentence to just talk about the notion of being able to use frozen Python modules to handle startup configuration tasks.